stable_baselines/common/vec_env/vec_frame_stack.py

```diff
@@ -24,9 +24,14 @@ def __init__(self, venv, n_stack):
 
     def step_wait(self):
         observations, rewards, dones, infos = self.venv.step_wait()
-        self.stackedobs = np.roll(self.stackedobs, shift=-observations.shape[-1], axis=-1)
+        last_ax_size = observations.shape[-1]
+        self.stackedobs = np.roll(self.stackedobs, shift=-last_ax_size, axis=-1)
         for i, done in enumerate(dones):
             if done:
+                old_terminal = infos[i]['terminal_observation']
```

Comment:
Please add a check here, and I don't know if we should warn or throw an exception if the key 'terminal_observation' is not present, what do you think @AdamGleave ?